### PR TITLE
Cleanup docstrings, add notebook render

### PR DESCRIPTION
**Pull request by CBroz1**
### Submission checklist

- [x] Reviewed all `Files changed` prior to selecting `Create Pull Request`.
- [X] Ran and passed all pytests, and ran the pipeline for features not covered by tests.
  - Caveat: there are pieces of the pytests needed adjustment to pass, including removing the 'ingest_lab' import
- [X] Ran pre-commit checks, including black, flake8, etc.
- [X] Updated the changelog and if making a new release, updated the version.py. **no new release**
- [X] Updated documentation, including concepts.md, tutorials.md, and relevant docstrings.
- [X] After merge, I will push a new tag to upstream (test* prefix if no new release).

### Submission details

Please review commits individually. Linting edits have been separated from substantive edits by commit

In this pull request, I make the following changes:
- Spellchecking
  - `cspell.json` - add permitted words for vscode spell extension 
  - `docs/mkdocs.yaml` - spellcheck notes
  - `docs/src/concepts` - remove singular bullets, add spaces before/after headings
  - various ephys schemas -  `scehma` -> `schema`
  - `element_array_ephys/plotting/qc.py` - various comment/docstring misspellings 
- Markdown linting
  - `.markdownlint.yaml` - config for markdown linting
  - `docs/src/concepts.md` - hard wrap, and change citation to footnote
  - `code_of_conduct` - ensure first line is heading 1
  -  hard wrap: `docs/src/index.md`, `docs/src/citation.md`, `contributing`, `changelog`
- Add tutorial notebooks
  - `gitignore` - add ignore of notebooks
  - `tutorials.md` -> `tutorials/index.md` - add summary of each notebook and links to other resources
  - `docs/src/api/make_pages.py` - migrate workflow download to docker compose
  - `docs/mkdocs.yaml` - add page for each notebook
  - `docs/src/citation.md` - change link to new tutorials index
  - `docs/.docker/pip_requirements` - add jupyter extension
- NWB maintenance
  - #123: `nwb_conversion_tools` -> `neuroconv`
  - migrate docstring style from sphynx to google for API docs
  - add `extras_require` installation options for nwb and dev work
- Other maintenance
  - #66 
  - `docs/docker-compose.yaml` - add mkdocs log file (gitignored) - tee pulls copy to log without silencing stdout
  - `docs/mkdocs.yaml` - add mkdocstrings params so api renders in file order, not alphabetically
  - `docs/mkdocs.yaml` - add footnotes extension for concepts page
  - `docs/src/changelog` - refresh as soft link
  - various ephys schemas - blackify, add argument descriptions for API docs, add return type suggestion for API docs
  - update license for 2023, datajoint neuro -> datajoint

### CODE REVIEW

File: element_array_ephys/probe.py [hunk 1]
```diff
@@ -216,7 +217,7 @@ def build_electrode_layouts(
     shank_count: int = 1,
     shank_spacing: float = None,
     y_origin="bottom",
-) -> list[dict]:
+) -> List[dict]:
```
Comment (JaerongA) [inline element_array_ephys/probe.py:220]:
```suggestion
) -> list[dict]:
```

File: element_array_ephys/probe.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 """
 Neuropixels Probes
 """
+from typing import List
```
Comment (JaerongA) [inline element_array_ephys/probe.py:4]:
```suggestion
```

**Comment by CBroz1** [comment]
Thanks @JaerongA for looking into typing compatibility. Thanks for running the pytests with the previous `list[dict]` format. I've applied your suggestions

**Review by JaerongA — APPROVED** [review]

**Review by ttngu207 — APPROVED** [review]

File: docs/src/concepts.md [hunk 3]
```diff
@@ -3,12 +3,28 @@
 ## Acquisition Tools for Electrophysiology
 
 ### Neuropixels Probes
-Neuropixels probes were developed by a collaboration between HHMI Janelia, industry partners, and others<sup>[1](#references)</sup>. Since their initial release in October 2018, 300 labs have ordered 1200 probes. Since the rollout of Neuropixels 2.0 in October 2020, IMEC has been shipping 100+ probes monthly (correspondence with Tim Harris).
 
-Neuropixels probes offer 960 electrode sites along a 10mm long shank, with 384 recordable channels per probe that can record hundreds of units spanning multiple brain regions (Neuropixels 2.0 version is a 4-shank probe with 1280 electrode sites per shank). Such large recording capacity has offered tremendous opportunities for the field of neurophysiology research, yet this is accompanied by an equally great challenge in terms of data and computation management.
+Neuropixels probes were developed by a collaboration between HHMI Janelia, industry
+partners, and others[^1]. Since their initial release in October
```
Comment (sidhulyalkar) [inline docs/src/concepts.md:8]: Is this citation link supposed to link to anything, or does it just provide reference to the citations located elsewhere?

Comment (CBroz1) [reply docs/src/concepts.md:8]: @sidhulyalkar `[^1]` is the format for the mkdocs footnotes extension. The citation is provided on line 19 below. [Example deployment](https://datajoint.com/docs/elements/element-electrode-localization/0.1/concepts/)

**Review by sidhulyalkar — APPROVED** [review]

**Pull request merged**